content/en/docs/2023.11/Reference/Blocks/Text/find-text/find-all-text-block.md

```diff
@@ -0,0 +1,579 @@
+          }
+        ]
+      }
+    }
+  }
+]
+```
+
+***
+
+### Find all occurrences that start with and end with a text in Text To Find (Null contains)
+
+This example will find all occurrences of text that start with `"The"`, contains `null` and ends with `"jumps"` from `"The quick brown fox jumps over the lazy dog. The dog woke up and tried to bite the fox. The fox jumps to get away."`.
+
+It performs a [case-sensitive, culture-insensitive][Ordinal] comparison of text.
+
+#### Properties
+
+| Property           | Value                     | Notes                                    |
+|--------------------|---------------------------|------------------------------------------|
+| [Text][Text Property] | `($)Text`, with value `"The quick brown fox jumps over the lazy dog. The dog woke up and tried to bite the fox. The fox jumps to get away."` | `($)Text` is a variable of type [String][] |
+| [Text To Find][TextToFind Property] | `($)TextToFind`, with value `{"startsWith": "The", "contains": null, "endsWith": "jumps"}`<br><br>In this example `($)TextToFind` has been set up using the following [Expression][]:<br><br>`new TextToFind(startsWith: "The", contains: null, endsWith:"jumps")` | `($)TextToFind` is a variable of type [TextToFind][] |
+| [Search Options][SearchOptions Property] | `($)SearchOptions`, with value `SearchOptions.ContainsText` | `($)SearchOptions` is a variable of type [SearchOptions][] |
+| [Comparison Type][ComparisonType Property] | `($)ComparisonType`, with value `StringComparison.Ordinal` | `($)ComparisonType` is a variable of type [StringComparison][] |
+| [Matches][Matches Property] | `($)Matches`, with no value | `($)Matches` is a variable that will be set to an [List][]<[Match][]> value |
+
+#### Result
+
+`"The quick brown fox jumps over the lazy dog. The dog woke up and tried to bite the fox. The fox jumps to get away."` has 0 sequences starting with `"The"` and ending with `"jumps"`, as [Contains][] being `null` will require an exact match for [Text to Find][TextToFind Property] in [Text][Text Property]. Therefore, the variable `($)Matches` will be set to the following:
+
+```json
+[]
+```
+
+***
+
+## Properties
+
+### Text
+
+The [Text][Text Property] to findall occurrences of [Text To Find][TextToFind Property] in.
+
+| | |
+|--------------------|---------------------------|
+| Data Type | [String][] |
+| Property Type | [InputOutput][] |
+| Is [Advanced][] | `false` |
+| Default Editor | [Variable][] |
+| Default Value | `($)Text` with no value |
+
+### Text To Find
+
+The [Text To Find][TextToFind Property] search query to find all occurrences of in [Text][Text Property]. This property contains all of the information in relation to the conditions for a valid match; these are:
+
+* [Starts With][StartsWith]
+* [Contains][Contains]
+* [Ends With][EndsWith]
+
+| | |
+|--------------------|---------------------------|
+| Data Type | [TextToFind][] |
+| Property Type | [Input][] |
+| Is [Advanced][] | `false` |
```

Comment:
Although the main property isn't advanced. It might be worth putting an extra note on this stating that the StartsWith and EndsWith sub-properties are advanced.